CMakeLists.txt

```diff
@@ -245,68 +230,63 @@ add_library(
 
 #-------------------------------------------------------------------------------
 # CUDA support.
+message( "" )
 set( blaspp_use_cuda false )  # output in blasppConfig.cmake.in
 if (gpu_backend MATCHES "^(auto|cuda)$")
-    include( CheckLanguage )
-    check_language( CUDA )
-    if (gpu_backend STREQUAL "cuda" AND NOT CMAKE_CUDA_COMPILER)
-        message( FATAL_ERROR "gpu_backend = ${gpu_backend}, but CUDA was not found")
+    message( STATUS "${bold}Looking for CUDA${not_bold} (gpu_backend = ${gpu_backend})" )
+    if (gpu_backend STREQUAL "cuda")
+        find_package( CUDAToolkit REQUIRED )
+    else()
+        find_package( CUDAToolkit QUIET )
+    endif()
+    if (CUDAToolkit_FOUND)
+        set( gpu_backend "cuda" )
+        set( blaspp_defs_cuda_ "-DBLAS_HAVE_CUBLAS" )
+        set( blaspp_use_cuda true )
+
+        # Some platforms need these to be public libraries.
+        target_link_libraries(
+            blaspp PUBLIC CUDA::cudart CUDA::cublas )
+        message( STATUS "${blue}Building CUDA support${plain}" )
+    else()
+        message( STATUS "${red}No CUDA support: CUDA not found${plain}" )
     endif()
 else()
-    message( STATUS "Skipping CUDA search. gpu_backend = ${gpu_backend}" )
-endif()
-if (CMAKE_CUDA_COMPILER)
-    enable_language( CUDA )
-
-    # todo: check for cuda 10.0+ and force newer CMake version - if possible
-
-    # CMake 3.17 adds find_package( CUDAToolkit )
-    # with CUDA::cudart and CUDA::cublas targets.
-    # For compatibility with older CMake, for now do search ourselves.
-    find_library( cudart_lib cudart ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES} )
-    find_library( cublas_lib cublas ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES} )
-
-    set( blaspp_defs_cuda_ "-DBLAS_HAVE_CUBLAS" )
-    set( blaspp_use_cuda true )
-
-    # Some platforms need these to be public libraries.
-    target_link_libraries(
-        blaspp PUBLIC "${cudart_lib}" "${cublas_lib}" )
-    target_include_directories(
-        blaspp PUBLIC "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}" )
-    message( STATUS "Building CUDA support in BLAS++" )
-else()
-    message( STATUS "Not building CUDA support in BLAS++" )
+    message( STATUS "${red}No CUDA support: gpu_backend = ${gpu_backend}${plain}" )
```

Comment:
I am not sure I understand this message. When the `gpu_backend` is not cuda, this message seems a little odd, is not it?

Reply:
It gives the user a reason why there is no CUDA support. It might say
`No CUDA support: gpu_backend = hip`
or
`No CUDA support: gpu_backend = none`
If a user emails us asking why it doesn't work, we can see immediately what went wrong.